reproduce-figures/README.md

```diff
@@ -1,10 +1,75 @@
+  * Note that this script expects these TSV files to be present in this directory; please do not move them.
+
+**Caution: This process script requires at least 170 GB free storage space.**
 
 ## Instructions
 
-_Forthcoming._
+To reproduce figures and analyses in this repository, you will need to organize and prepare all data needed for input.
+Follow the steps below to do so:
+
+### Step 1: Set up the `renv` environment
+
+This repository uses [`renv`](https://rstudio.github.io/renv/) to manage R packages and dependencies.
+To set up the `renv` environment, open the the `scpca-paper-figures.Rproj` project in RStudio and run `renv::restore()` to install all packages in the provided `renv.lock` file.
+Note that this project uses R version `4.4.0`.
+
+Please refer to the [Introduction to `renv` vignette](https://rstudio.github.io/renv/articles/renv.html) for more information about getting started with `renv`.
+
+### Step 2: Download files from the ScPCA Portal
+
+You will need to download the following files from the ScPCA Portal <https://scpca.alexslemonade.org/>:
+
+1. All ScPCA projects listed in the [Project Whitelist file](../sample-info/project-whitelist.txt)
+  * On each project page (<https://scpca.alexslemonade.org/projects/SCPCPXXXXXX>), download each project with the following options selected:
+    * Ensure the selected modality is `Single-cell`
+    * Ensure the selected Data Format is `SingleCellExperiment (r)`
+    * Do _not_ click to merge all objects into the same sample
+    * If the project contains multiplexed samples, you can exclude those samples (but it will not affect the code if they are included)
+  * These projects will download as zip files; you should leave them in this compressed format, and _do not rename them_.
+
+2. The portal-wide metadata TSV file
+  * You can download this single TSV file using the "Get All Sample Metadata" button on the top-right of the portal homepage.
+
+
+### Step 3: Organize your downloaded files
+
+Place all downloaded ScPCA project files are in the same directory.
+Optionally, you can also store the portal metadata TSV in this directory, but it is not necessary.
+
+### Step 4: Run the `prepare-scpca-portal-data.R` script
+
+Now, you can run the `prepare-scpca-portal-data.R` script specifying the following input arguments:
+
+* `--portal_projects_dir`: Path to the directory containing the ScPCA project zip files
+* `--portal_metadata_path`: Path to the portal-wide metadata TSV file
+
+```sh
+  Rscript prepare-scpca-portal-data.R \
+    --portal_projects_dir <path to directory with all ScPCA project zip files> \
+    --portal_metadata_path <path to portal-wide metadata TSV>
+```
+
+This script will run for roughly 90 minutes, and it will create three directories within this repository with files organized as code expects:
+
+* `../s3_files`: This directory will contain ScPCA data, metadata, and reference files needed to reproduce figures, tables, and the bulk expression analysis
```

Comment:
What files are in`s3_files` that are used for bulk analysis?

Reply:
Consensus cell types are used to make the genesets we use in the over-representation analysis.
That said, these are all in the repository - https://github.com/AlexsLemonade/scpca-paper-figures/tree/main/analysis/pseudobulk-bulk-prediction/data
We could make this an optional step in the bulk pipeline if those exist, but I don't think we need to.

Reply:
Ah yes! I was just curious since looking at the folder nothing was obvious to me so I just wanted to double check your statement was correct. No need to make any additional code changes!